go/arrow/ipc/file_reader.go

```diff
@@ -348,10 +356,26 @@ func (src *ipcSource) buffer(i int) *memory.Buffer {
 		return memory.NewBufferBytes(nil)
 	}
 
-	raw := make([]byte, buf.Length())
-	_, err := src.r.ReadAt(raw, buf.Offset())
-	if err != nil {
-		panic(err)
+	var raw []byte
+	if src.codec == nil {
+		raw = make([]byte, buf.Length())
+		_, err := src.r.ReadAt(raw, buf.Offset())
+		if err != nil {
+			panic(err)
```

Comment:
seems like panic is used in existing code, this seems strange to me though.

Reply:
the choice to use panic is probably to make the consumable api easier to manage instead of having to propagate and push errors everywhere it just fails hard. Since the existing code was already using panic, that's what I maintained in my changes rather than adding error out params.

Reply:
seems OK to me.  I don't know Go well enough but for IO related code I would expect failures at least some of the time and I thought panic was meant for actual "stop the entire program" type errors.

Reply:
`panic` is relatively equivalent to throwing an exception in C++ code. You can recover from a panic just like you could catch a thrown exception in C++ code. The "general" advice between returning an error vs throwing a panic is whether or not the error happened during *normal* operation or happened due to something being coded/programmed incorrectly (like a pre/post-condition being violated etc.) 

As i mentioned, normally *I* would probably return the error here rather than panic, but I was maintaining the current pattern that existed in the code rather than creating a breaking change which would make consumers have to update their call-sites to handle the extra error out param.